### PR TITLE
Add docs directory to template

### PR DESCRIPTION
**Pull request by charlieduong94**
Added stub development, data model, and authentication docs
to the template. The goals is to encourage developers to write
documentation as they build out integrations.

### CODE REVIEW

**Comment by charlieduong94** [comment]
Once this is approved, I'll update the https://github.com/JupiterOne/graph-snowflake and https://github.com/JupiterOne/graph-trend-micro projects to include these files.

File: template/docs/authentication.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Authentication
```
Comment (erkangz) [inline template/docs/authentication.md:1]: can we combine Authentication and Data Model into a single file? That would serve as a user-facing integration doc that we can easily include in `docs` to publish to support site.

Comment (charlieduong94) [reply template/docs/authentication.md:1]: Sure. We can do that. What do you think about placing that under perhaps an `integration.md` file?

Comment (charlieduong94) [reply template/docs/authentication.md:1]: or maybe `jupiterone.md`?

**Pull request merged**